SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -34,6 +34,8 @@ @interface SDLIAPTransport () {
 @property (assign) int retryCounter;
 @property (assign) BOOL sessionSetupInProgress;
 @property (strong) SDLTimer *protocolIndexTimer;
+@property (nonatomic, assign) UIBackgroundTaskIdentifier backgroundTaskId;
+@property (assign) BOOL isDelayedConnect;
```

Comment:
Are both `backgroundTaskId` and `isDelayedConnect` needed? I really dislike state booleans, and any way to get rid of them and provide stateless code should be taken. They make unit testing extremely difficult. Is it possible to use the `backgroundTaskId` and check whether it's `invalid` to know if we're in a delayed connect?